Source/engine/point.hpp

```diff
@@ -12,99 +13,81 @@
 
 namespace devilution {
 
-struct Point {
-	int x;
-	int y;
+template <typename CoordT>
+struct PointOf;
 
-	Point() = default;
+using Point = PointOf<int>;
 
-	constexpr Point(int x, int y)
+template <typename CoordT>
+struct PointOf {

```

Comment:
How about this?

```suggestion
template <typename CoordT = int>
struct Point {
```

Same goes for `DisplacementOf`.

Reply:
For such types, `Point a;` is only legal in C++17. We're somewhere close to C++11 on the language support.